api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/advancedhivillness/ResumoMensalDAHCohortQueries.java

```diff
@@ -0,0 +1,349 @@
+    cd.addParameters(getCohortParameters());
+
+    cd.addSearch(
+        "onDAHDuringPeriod",
+        map(
+            listOfPatientsInAdvancedHivIllnessCohortQueries.getPatientsWhoStartedFollowupOnDAH(
+                true),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+
+    cd.addSearch(
+        "onArt",
+        map(
+            getPatientsArtSituationOnDAH(hivMetadata.getArtStatus()),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+
+    cd.addSearch(
+        "B12",
+        map(
+            resumoMensalCohortQueries.getPatientsWhoWereActiveByEndOfPreviousMonthB12(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+
+    cd.setCompositionString("onDAHDuringPeriod AND (onArt OR B12)");
+    return cd;
+  }
+
+  /**
+   *
+   * <li>Com registo de pelo menos um motivo (Óbito/ Abandono/ Transferido Para) e “Data de Saída de
+   *     TARV na US” (secção J), na Ficha de DAH, ocorrida após a data mais recente da “Data de
+   *     Início no Modelo de DAH” e até o fim do mês anterior [“Data de Saída de TARV na US” >=
+   *     “Última Data de Início no Modelo de DAH” e <= “Data Início” menos (-) 1 dia] ou
+   * <li>Com registo de “Data de Saída” (secção I), registada na Ficha de DAH e ocorrida após a data
+   *     mais recente da “Data de Início no Modelo de DAH” e até o fim do mês anterior [“Data de
+   *     Saída de TARV na US” >= “Última Data de Início no Modelo de DAH” e <= “Data Início” menos
+   *     (-) 1 dia]
+   *
+   * @return {@link CohortDefinition}
+   */
+  public CohortDefinition getPatientsWhoLeftFollowupOnDAHByTheEndOfPreviousMonth() {
+
+    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
+    sqlCohortDefinition.setName(
+        " utentes que saíram do seguimento para DAH até o fim do mês anterior");
+    sqlCohortDefinition.addParameter(new Parameter("startDate", "startDate", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("location", "location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("90", hivMetadata.getAdvancedHivIllnessEncounterType().getEncounterTypeId());
+    map.put("1366", hivMetadata.getPatientHasDiedConcept().getConceptId());
+    map.put("1706", hivMetadata.getTransferredOutConcept().getConceptId());
+    map.put("1707", hivMetadata.getAbandonedConcept().getConceptId());
+    map.put("1708", hivMetadata.getExitFromArvTreatmentConcept().getConceptId());
+    map.put("165386", hivMetadata.getExitDateFromArvTreatmentConcept().getConceptId());
+
+    String query =
+        "SELECT p.patient_id "
+            + "FROM "
+            + "    patient p INNER JOIN encounter e ON p.patient_id = e.patient_id "
+            + "              INNER JOIN obs o on e.encounter_id = o.encounter_id "
+            + "              INNER JOIN obs o2 on e.encounter_id = o2.encounter_id "
```

Comment:
@helderjosue You don't need this o2, it is going to slow your query

Reply:
Noted and Corrected!